avm/res/network/azure-firewall/main.bicep

```diff
@@ -278,44 +281,62 @@ resource azureFirewall 'Microsoft.Network/azureFirewalls@2024-05-01' = {
   zones: map(availabilityZones, zone => '${zone}')
   tags: tags
   properties: azureSkuName == 'AZFW_VNet'
-    ? {
-        threatIntelMode: threatIntelMode
-        firewallPolicy: !empty(firewallPolicyId)
-          ? {
-              id: firewallPolicyId
-            }
-          : null
-        ipConfigurations: ipConfigurations
-        managementIpConfiguration: requiresManagementIp ? managementIPConfiguration : null
-        sku: {
-          name: azureSkuName
-          tier: azureSkuTier
-        }
-        applicationRuleCollections: applicationRuleCollections ?? []
-        natRuleCollections: natRuleCollections ?? []
-        networkRuleCollections: networkRuleCollections ?? []
-      }
-    : {
-        autoscaleConfiguration: {
-          maxCapacity: autoscaleMaxCapacity
-          minCapacity: autoscaleMinCapacity
-        }
-        firewallPolicy: !empty(firewallPolicyId)
+    ? union(
```

Comment:
Would highly recommend to use `shallowMerge()` via `...(condition ? {  properties} : {})` instead of union to preserve linter support